### PR TITLE
TiledMap supports object and class type properties, and supports multi-line string properties. 

### PR DESCRIPTION
**Pull request by zhakesi**
this pr is contributed by  **biliskener**    migrated from
* https://github.com/cocos/cocos-engine/pull/13960

### CODE REVIEW

File: cocos/tiledmap/tiled-types.ts [hunk 1]
```diff
@@ -548,6 +548,8 @@ export class TMXImageLayerInfo {
     sourceImage?: SpriteFrame;
 
     tintColor: Color | null = null;
+
+    properties: PropertiesInfo = {} as any;
```
Comment (SantyWang) [inline cocos/tiledmap/tiled-types.ts:552]: Seems not be used!

File: cocos/tiledmap/tmx-xml-parser.ts [hunk 2]
```diff
@@ -89,15 +89,21 @@ function strToColor (value: string): Color {
...
+                    res.push(childNode2);
+                }
+            }
         }
     }
 
     map = map || ({} as any);
+    const className = node.getAttribute('class');
+    if (className) {
+        map!.propClass = className;
```
Comment (SantyWang) [inline cocos/tiledmap/tmx-xml-parser.ts:105]: Seems not be used!

File: cocos/tiledmap/tmx-xml-parser.ts [hunk 3]
```diff
@@ -112,6 +118,12 @@ function getPropertyList (node: Element, map?: PropertiesInfo): PropertiesInfo {
             value = value === 'true';
         } else if (type === 'color') {
             value = strToColor(value);
+        } else if (type === 'class') {
+            value = getPropertyList(element, {});
+        } else if (type === 'object') {
+            value = parseInt(value);
```
Comment (SantyWang) [inline cocos/tiledmap/tmx-xml-parser.ts:124]: Why parse object as int

**Comment by SantyWang** [comment]
Shoule refine the api docs before releasing this feature

**Comment by zhakesi** [comment]
will be merged in next version

**Pull request closed without merging**